scripts/mailmap-update.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -eu
+
+usage () {
+    echo "USAGE: ./update-mailmap.sh <bpftool-repo> <linux-repo>"
+    exit 1
+}
+
+LIBBPF_REPO="${1-""}"
+LINUX_REPO="${2-""}"
+
+if [ -z "${LIBBPF_REPO}" ] || [ -z "${LINUX_REPO}" ]; then
+    echo "Error: libbpf or linux repos are not specified"
+    usage
+fi
+
+LIBBPF_MAILMAP="${LIBBPF_REPO}/.mailmap"
+LINUX_MAILMAP="${LINUX_REPO}/.mailmap"
+
+tmpfile="$(mktemp)"
+cleanup() {
+    rm -f "${tmpfile}"
+}
+trap cleanup EXIT
+
+grep_lines() {
+    local pattern="$1"
+    local file="$2"
+    grep "${pattern}" "${file}" || true
+}
+
+while read -r email; do
+    grep_lines "${email}$" "${LINUX_MAILMAP}" >> "${tmpfile}"
+done < <(git log --format='<%ae>' | sort -u)
```

Comment:
is this going to do a full history log? that's quite slow. Maybe let's log only newly synced commits, and then merge new authors and those that are in .mailmap? That would be much faster, right?

Reply:
We can, although the operation doesn't seem to take that long on my laptop:

```console
$ time (git log --format='<%ae>' | sort -u > /dev/null)
( git log --format='<%ae>' | sort -u > /dev/null; )  0.05s user 0.01s system 118% cpu 0.047 total
```

I can look into it in the coming days.

Reply:
ah, this is supposed to be run from libbpf repo, not linux repo, right? I see, ok, let's keep it simple for now.

Reply:
Thinking again about this: If we were to do incremental updates, we could miss new entries in Linux's mailmap.

For example:

- Someone, contributor C, contributed their changes in libbpf v1.4
- Contributor C does not send new contributions to libbpf after v1.4 has been tagged
- We get more contributions, sync the repo...
- Meanwhile, contributor C adds an entry for their email in Linux's .mailmap
- We're updating libbpf's .mailmap, considering only the contributions since the latest sync, say since v1.4: There's no contributions from contributor C on that period, so we're not looking for their email in Linux's .mailmap. We're not adding any entry (although one exists on Linux's side) and we don't reflect the email or name change in `git log`.

Reply:
Yes, but libbpf log keeps growing over time and if it starts taking more time just to list all commits we'd need to think about this problem some more. For now it's not a big deal, so let's not fix what's not broken.

Reply:
Agreed. We should be good for some time: the command takes me 0.3s in Cilium's repo with its 28k+ commits.